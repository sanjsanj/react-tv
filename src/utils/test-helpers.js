export const messagesPayload = [
  {
    id: 'cd445e6d-e514-424f-ba8f-16ec842002c6',
    userId: 'fe27b760-a915-475c-80bb-7cdf14cc6ef3',
    message: 'Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla.',
    timestamp: '2017-02-09T04:27:38Z',
  },
  {
    id: 'b03569ae-ccbf-4975-8040-4daba638b407',
    userId: '16373df5-da0a-4074-8295-f916b94269f4',
    message: 'Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.',
    timestamp: '2016-11-09T05:04:58Z',
  },
];

export const membersPayload = [
  {
    id: 'fe27b760-a915-475c-80bb-7cdf14cc6ef3',
    firstName: 'Martin',
    lastName: 'Bradley',
    email: 'mbradley0@google.it',
    avatar: 'http://dummyimage.com/100x100.png/5fa2dd/ffffff',
    ip: '166.124.172.160',
  },
  {
    id: '16373df5-da0a-4074-8295-f916b94269f4',
    firstName: 'Helen',
    lastName: 'Hawkins',
    email: 'hhawkins1@posterous.com',
    avatar: 'http://dummyimage.com/100x100.jpg/dddddd/000000',
    ip: '179.239.189.173',
  },
];
