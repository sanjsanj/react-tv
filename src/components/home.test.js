import React from 'react';
import { shallow } from 'enzyme';

import { messagesPayload, membersPayload } from '../utils/test-helpers';

import Home from './home';

describe('Home', () => {
  let messageList;
  let props;
  let tree;

  beforeAll(() => {
    props = {
      messages: messagesPayload,
      members: membersPayload,
      loadMessages: jest.fn(),
      loadMembers: jest.fn(),
    };

    tree = shallow(<Home.WrappedComponent {...props} />);
    messageList = tree.find('Message');
  });

  it('should match snapshot', () => {
    expect(tree).toMatchSnapshot();
  });

  it('should display 2 messages', () => {
    expect(messageList.length).toEqual(2);
  });

  it('should sort messages by time descending', () => {
    const unorderedMessageProps = {
      messages: [messagesPayload[1], messagesPayload[0]],
      members: membersPayload,
      loadMessages: jest.fn(),
      loadMembers: jest.fn(),
    };

    tree = shallow(<Home.WrappedComponent {...unorderedMessageProps} />);

    const newestMessageData = tree.childAt(0).prop('data');

    expect(newestMessageData).toEqual(messagesPayload[0]);
  });
});
