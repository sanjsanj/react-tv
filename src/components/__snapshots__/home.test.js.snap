// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
<ul>
  <Message
    data={
      Object {
        "id": "cd445e6d-e514-424f-ba8f-16ec842002c6",
        "message": "Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla.",
        "timestamp": "2017-02-09T04:27:38Z",
        "userId": "fe27b760-a915-475c-80bb-7cdf14cc6ef3",
      }
    }
    key="cd445e6d-e514-424f-ba8f-16ec842002c6"
    user={
      Object {
        "avatar": "http://dummyimage.com/100x100.png/5fa2dd/ffffff",
        "email": "mbradley0@google.it",
        "firstName": "Martin",
        "id": "fe27b760-a915-475c-80bb-7cdf14cc6ef3",
        "ip": "166.124.172.160",
        "lastName": "Bradley",
      }
    }
  />
  <Message
    data={
      Object {
        "id": "b03569ae-ccbf-4975-8040-4daba638b407",
        "message": "Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.",
        "timestamp": "2016-11-09T05:04:58Z",
        "userId": "16373df5-da0a-4074-8295-f916b94269f4",
      }
    }
    key="b03569ae-ccbf-4975-8040-4daba638b407"
    user={
      Object {
        "avatar": "http://dummyimage.com/100x100.jpg/dddddd/000000",
        "email": "hhawkins1@posterous.com",
        "firstName": "Helen",
        "id": "16373df5-da0a-4074-8295-f916b94269f4",
        "ip": "179.239.189.173",
        "lastName": "Hawkins",
      }
    }
  />
</ul>
`;
